controller-api/src/main/java/com/yahoo/vespa/hosted/controller/api/identifiers/Identifier.java

```diff
@@ -12,7 +12,7 @@
  */
 public abstract class Identifier {
 
-    protected static final Pattern strictPattern = Pattern.compile("[a-z0-9][a-z0-9-]{0,26}[a-z0-9]");
+    protected static final Pattern strictPattern = Pattern.compile("^(?=.{1,28}$)[a-z0-9](-?[a-z0-9]+)*$");
```

Comment:
Explanatory comment could be useful, if I got this right it's
1) max 28 characters in length (why exactly 28?)
2) only a-z, 0-9 and single dash